src/data/list/basic.lean

```diff
@@ -2945,6 +2945,25 @@ theorem map_filter_map_of_inv (f : α → option β) (g : β → α)
   map g (filter_map f l) = l :=
 by simp only [map_filter_map, H, filter_map_some]
 
+theorem map_filter_map_some_eq_filter_map_is_some (f : α → option β) (l : list α) :
+  (l.filter_map f).map some = (l.map f).filter (λ b, b.is_some) :=
+begin
+  induction l with x xs ih,
+  { simp },
+  { cases h : f x; rw [list.filter_map_cons, h]; simp [h, ih] },
+end
+
+theorem length_filter_le (p : α → Prop) [decidable_pred p] (l : list α) :
+  (l.filter p).length ≤ l.length :=
```

Comment:
This belongs near the lemmas about filter, rather than near the filter_map lemmas

Reply:
Moving `length_filter_le` to the part about `filter` would require `section filter` to precede the part about `filter_map` but there already exist two dependencies in the other direction. However, one of them is `map_filter` which could be moved to the part about `filter_map`. What do you recommend?

Reply:
Don't worry about this, it's harder to move this lemma than I thought.